rust/projects/file-io/project.md

```diff
@@ -187,22 +236,164 @@ to read about the "get" command. It may help to keep both in mind, or to
 implement them both simultaniously. Your choice.
 
 
-## Part 3: Reading from the write-ahead log
+## Part 3: Reading from the log
+
+Now it's time to implement "get". It's again pretty straightforward in concept:
+just read each command in the log on startup, executing them to modify the
+in-memory index. Then read from the cache.
+
+Should you read all records in the log into memory at once and then replay
+them into your map type; or should you read them one at a time while
+replaying the into your map? Should you read into a buffer before deserializing
+or deserialize from a file stream? Think about the memory usage of your approach.
+Think about the way reading from I/O streams interacts with thet kernel.
+
+Remember that "get" may not find a value and that case has to be handled
+specially. Here, our API returns `None` and our command line client prints
+a particular message and exits with a non-zero exit code.
+
+There's one complication to reading the log, and you may have already considered
+it while writing thet "set" code: how do you distinguish between each record in
+the log? That is, how do you know when to stop reading one record, and start
+reading the next? Do you even need to? Maybe serde will deserialize a record
+directly from an I/O stream and stop reading when it's done, leaving the
+[cursor] in the correct place to read subsequent records. Maybe serde will
```

Comment:
What will "cursor" link to?

Reply:
"file cursor" on wikipedia. But instead I'll just say "file cursor" and our audience will probably know what that is.

Reply:
Fixed.